addons/account/models/account_move.py

```diff
@@ -1239,7 +1239,7 @@ def _compute_invoice_taxes_by_group(self):
             done_taxes = set()
             for line in tax_lines:
                 res.setdefault(line.tax_line_id.tax_group_id, {'base': 0.0, 'amount': 0.0})
-                res[line.tax_line_id.tax_group_id]['amount'] += line.price_subtotal
+                res[line.tax_line_id.tax_group_id]['amount'] += line.currency_id and line.amount_currency or line.price_subtotal
```

Comment:
There is something wrong here:
balance = debit - credit (company's currency)
amount_currency = balance in foreign currency
price_subtotal = signed(amount_currency or balance) depending the invoice's currency and the invoice's type.

In your case, `line.price_subtotal` is correct.